server/src/internalClusterTest/java/org/elasticsearch/monitor/metrics/NodeIndexingMetricsIT.java

```diff
@@ -0,0 +1,264 @@
+        assertBusy(() -> {
+            var coordinatingOperationsRejectionsTotal = getRecordedMetric(
+                plugin::getLongAsyncCounterMeasurement,
+                "es.indexing.coordinating_operations.rejections.total"
+            );
+            assertThat(coordinatingOperationsRejectionsTotal.getLong(), equalTo(1L));
+        });
+    }
+
+    public void testPrimaryRejectionMetricsArePublishing() throws Exception {
+
+        // setting low Indexing Pressure limits to trigger primary rejections
+        final String dataNode = internalCluster().startNode(Settings.builder().put(MAX_INDEXING_BYTES.getKey(), "1KB").build());
+        // setting high Indexing Pressure limits to pass coordinating checks
+        final String coordinatingNode = internalCluster().startCoordinatingOnlyNode(
+            Settings.builder().put(MAX_INDEXING_BYTES.getKey(), "10MB").build()
+        );
+        ensureStableCluster(2);
+
+        final TestTelemetryPlugin plugin = internalCluster().getInstance(PluginsService.class, dataNode)
+            .filterPlugins(TestTelemetryPlugin.class)
+            .findFirst()
+            .orElseThrow();
+        plugin.resetMeter();
+
+        final int numberOfShards = randomIntBetween(1, 5);
+        assertAcked(prepareCreate("test", Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, numberOfShards)).get());
+
+        final BulkRequest bulkRequest = new BulkRequest();
+        final int batchCount = randomIntBetween(50, 100);
+        for (int i = 0; i < batchCount; i++) {
+            bulkRequest.add(new IndexRequest("test").source("field", randomAlphaOfLength(2048)));
+        }
+
+        // big batch should pass thru coordinating limit check but fail on primary
+        // note the bulk request is sent to coordinating client
+        final BulkResponse bulkResponse = client(coordinatingNode).bulk(bulkRequest).actionGet();
+        assertThat(bulkResponse.hasFailures(), equalTo(true));
+        assertThat(Arrays.stream(bulkResponse.getItems()).allMatch(item -> item.status() == RestStatus.TOO_MANY_REQUESTS), equalTo(true));
+
+        // simulate async apm `polling` call for metrics
+        plugin.collect();
+
+        // this bulk request is too big to pass coordinating limit check
+        assertBusy(() -> {
+            var primaryOperationsRejectionsTotal = getRecordedMetric(
+                plugin::getLongAsyncCounterMeasurement,
+                "es.indexing.primary_operations.rejections.total"
+            );
+            assertThat(primaryOperationsRejectionsTotal.getLong(), equalTo((long) numberOfShards));
+        });
+
+    }
+
+    private static Measurement getRecordedMetric(Function<String, List<Measurement>> metricGetter, String name) {
+        final List<Measurement> measurements = metricGetter.apply(name);
+        assertFalse("Indexing metric is not recorded", measurements.isEmpty());
+        assertThat(measurements.size(), equalTo(1));
```

Comment:
Consider using `assertThat(measurements, hasSize(1));`. In case of failure the assertion error message will contain the collection contents making it easier to understand what is extra.